In [1]:
import pandas as pd

In [2]:
import numpy as np

In [ ]:
#pip install openpyxl

In [ ]:
#!pip install --upgrade_pandas

In [3]:
print(pd.__version__)

1.5.3


In [4]:
!pip install xlrd

  Obtaining dependency information for xlrd from https://files.pythonhosted.org/packages/1a/62/c8d562e7766786ba6587d09c5a8ba9f718ed3fa8af7f4553e8f91c36f302/xlrd-2.0.2-py2.py3-none-any.whl.metadata
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 kB 3.1 MB/s eta 0:00:00


In [ ]:
#!conda install -c anaconda xlrd -y

In [5]:
pf = pd.read_excel("GSAF5.xls")

In [6]:
pf

Date    Year        Type             Country  \
0       14th October  2025.0  Unprovoked            Columbia   
1       11th October  2025.0  Unprovoked           Australia   
2        7th October  2025.0  Unprovoked           Australia   
3     29th September  2025.0  Unprovoked                 USA   
4     27th September  2025.0    Provoked          Costa Rica   
...              ...     ...         ...                 ...   
7045     Before 1903     0.0  Unprovoked           AUSTRALIA   
7046     Before 1903     0.0  Unprovoked           AUSTRALIA   
7047       1900-1905     0.0  Unprovoked                 USA   
7048       1883-1889     0.0  Unprovoked              PANAMA   
7049       1845-1853     0.0  Unprovoked  CEYLON (SRI LANKA)   

                      State                             Location  \
0      Bolivar, del Isolate                    Catagena Province   
1                Queensland       Cook Esplanade Thursday Island   
2           South Australia                      Kangaroo Island   
3            Off California                      Catalina Island   
4                       NaN                        Cocos Islands   
...                     ...                                  ...   
7045      Western Australia                          Roebuck Bay   
7046      Western Australia                                  NaN   
7047         North Carolina                       Ocracoke Inlet   
7048                    NaN                 Panama Bay 8ºN, 79ºW   
7049       Eastern Province  Below the English fort, Trincomalee   

                   Activity                   Name Sex  Age  ...  \
0      Swimming with sharks             Male child   M   14  ...   
1          Fishing/swimming             Samuel Nai   M   14  ...   
2                   Surfing           Lee Berryman   M  50+  ...   
3                  Swimming     Christopher Murray   M   54  ...   
4     Diving-Tagging sharks     Dr. Mauricio Hoyos   M   48  ...   
...                     ...                    ...  ..  ...  ...   
7045                 Diving                   male   M  NaN  ...   
7046           Pearl diving                  Ahmun   M  NaN  ...   
7047               Swimming  Coast Guard personnel   M  NaN  ...   
7048                    NaN        Jules Patterson   M  NaN  ...   
7049               Swimming                   male   M   15  ...   

                 Species                                          Source  \
0            Nurse shark   Kevin McMurray Trackingsharks.com Andy Currie   
1     Tiger or Bull shark              Kevin McMurray Trackingsharks.com   
2          Bronze whaler?              Kevin McMurray Trackingsharks.com   
3      unknown 1.2m shark  Todd Smith: Kevin McMurray Trackingsharks.com   
4          Tiger shark 4m  Todd Smith: Kevin McMurray Trackingsharks.com   
...                   ...                                            ...   
7045                  NaN               H. Taunton; N. Bartlett,  p. 234   
7046                  NaN          H. Taunton; N. Bartlett,  pp. 233-234   
7047                  NaN           F. Schwartz, p.23; C. Creswell, GSAF   
7048                  NaN                            The Sun, 10/20/1938   
7049                  NaN                                     S.W. Baker   

                                 pdf  \
0                                NaN   
1                                NaN   
2                                NaN   
3                                NaN   
4                                NaN   
...                              ...   
7045          ND-0005-RoebuckBay.pdf   
7046               ND-0004-Ahmun.pdf   
7047  ND-0003-Ocracoke_1900-1905.pdf   
7048      ND-0002-JulesPatterson.pdf   
7049              ND-0001-Ceylon.pdf   

                                           href formula  \
0                                                   NaN   
1                                                   NaN   
2                                         

In [7]:
pf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7050 entries, 0 to 7049
Data columns (total 23 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Date            7050 non-null   object 
 1   Year            7048 non-null   float64
 2   Type            7032 non-null   object 
 3   Country         7000 non-null   object 
 4   State           6564 non-null   object 
 5   Location        6483 non-null   object 
 6   Activity        6465 non-null   object 
 7   Name            6831 non-null   object 
 8   Sex             6471 non-null   object 
 9   Age             4056 non-null   object 
 10  Injury          7015 non-null   object 
 11  Fatal Y/N       6489 non-null   object 
 12  Time            3524 non-null   object 
 13  Species         3919 non-null   object 
 14  Source          7030 non-null   object 
 15  pdf             6799 non-null   object 
 16  href formula    6794 non-null   object 
 17  href            6796 non-null   o

### Problem Statements:

Explenation: A problem statement clearly describes the business issue to be addressed

It is not yet clear which environmental or behavioral factors most strongly contribute to shark attack frequency. The dataset must be cleaned and analyzed to determine whether activity type, location, or demographic characteristics play a significant role.

### Hypothesis:

Explenation: A hypothesis is a testable statement derived from the problem statement, providing a basis for investigation.

- Shark attacks are more common in certain countries.
- Shark attacks happen more often during surfing.







### Minimum Viable Product (MVP):

Explenation: An MVP is an initial version of a product containing only the fundamental features necessary to meet the essential needs of its first users. This approach allows for early feedback and iterative improvements. In data analytics, an MVP might be a preliminary model providing basic insights, with further optimization after feedback.

mvp version would need to include the following columns:
0   Date            7050 non-null   object 
 1   Year            7048 non-null   float64
 2   Type            7032 non-null   object 
 3   Country         7000 non-null   object 
 4   State           6564 non-null   object 
 5   Location        6483 non-null   object 
 6   Activity        6465 non-null   object 
 7   Name            6831 non-null   object 
 8   Sex             6471 non-null   object 
 9   Age             4056 non-null   object 
 10  Injury          7015 non-null   object 
 11  Fatal Y/N       6489 non-null   object 
 12  Time            3524 non-null   object 
 13  Species         3919 non-null   object 

Not include:
 14  Source          7030 non-null   object 
 15  pdf             6799 non-null   object 
 16  href formula    6794 non-null   object 
 17  href            6796 non-null   object 
 18  Case Number     6798 non-null   object 
 19  Case Number.1   6797 non-null   object 
 20  original order  6799 non-null   float64
 21  Unnamed: 21     1 non-null      object 
 22  Unnamed: 22     2 non-null      object 


### Project Scoping:

Explenation: Scoping defines the project’s boundaries, specifying objectives, required resources, key features, priorities, and constraints. It ensures team alignment and resource availability. By clear scoping, projects are more manageable and focused, reducing the risk of scope creep.

### Roadmapping:

Explenation: A roadmap lays out the project’s timeline and major milestones, from data collection to model deployment. It helps in setting expectations, managing stakeholders, and tracking progress. In data analytics, roadmaps might include stages such as data preprocessing, analysis, modeling, and reporting.

### 

Explenation:

### 

Explenation:

### 

Explenation: